src/ess/reduce/time_of_flight/interpolator_numba.py

```diff
@@ -0,0 +1,156 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025 Scipp contributors (https://github.com/scipp)
+import numpy as np
+from numba import njit, prange
+
+
+@njit(boundscheck=False, cache=True, fastmath=True, parallel=True)
+def interpolate(
+    x: np.ndarray,
+    y: np.ndarray,
+    z: np.ndarray,
+    values: np.ndarray,
+    xp: np.ndarray,
+    yp: np.ndarray,
+    zp: np.ndarray,
+    fill_value: float,
+    out: np.ndarray,
+):
+    """
+    Linear interpolation of data on a 3D regular grid.
+
+    Parameters
+    ----------
+    x:
+        1D array of grid edges along the x-axis. They must be linspaced.
+    y:
+        1D array of grid edges along the y-axis. They must be linspaced.
+    z:
+        1D array of grid edges along the z-axis. They must be linspaced.
+    values:
+        3D array of values on the grid. The shape must be (nz, ny, nx).
+    xp:
+        1D array of x-coordinates where to interpolate (size N).
+    yp:
+        1D array of y-coordinates where to interpolate (size N).
+    zp:
+        1D array of z-coordinates where to interpolate (size N).
+    fill_value:
+        Value to use for points outside of the grid.
+    out:
+        1D array where the interpolated values will be stored (size N).
+    """
+    npoints = len(xp)
+    xmin = x[0]
+    xmax = x[-1]
+    ymin = y[0]
+    ymax = y[-1]
+    zmin = z[0]
+    zmax = z[-1]
+    dx = x[1] - xmin
+    dy = y[1] - ymin
+    dz = z[1] - zmin
+
+    for i in prange(npoints):
+        xx = xp[i]
+        yy = yp[i]
+        zz = zp[i]
+
+        if (
+            (xx < xmin)
+            or (xx > xmax)
+            or (yy < ymin)
+            or (yy > ymax)
+            or (zz < zmin)
+            or (zz > zmax)
+        ):
+            out[i] = fill_value
+
+        else:
+            ix = int((xx - xmin) / dx)
+            iy = int((yy - ymin) / dy)
+            iz = int((zz - zmin) / dz)
```

Comment:
```suggestion
            ix = (xx - xmin) // dx
            iy = (yy - ymin) // dy
            iz = (zz - zmin) // dz
```
Or does numba bot support this? Or does it round in the wrong way?

Reply:
It should be fine with your suggestion. They are equivalent, aren't they?

Reply:
Not for negative numbers:
```python
>>> int(-3/2)
-1
>>> -3//2
-2
```

Reply:
Ah yes, but I don't think it matters in this case as we are always taking `(x - xmin)` which should always be positive or 0. If it is negative, it was caught by the `if xx < xmin` further up.